exec/java-exec/src/main/java/org/apache/calcite/jdbc/DynamicRootSchema.java

```diff
@@ -154,10 +157,25 @@ private void loadSchemaFactory(String schemaName, boolean caseSensitive) {
     }
   }
 
-  static class RootSchema extends AbstractSchema {
+  public static class RootSchema extends AbstractSchema {
 
-    public RootSchema() {
+    private StoragePluginRegistry storages;
+
+    private DynamicRootSchema dynRootSchema;
+
+    public RootSchema(StoragePluginRegistry storages) {
       super(Collections.emptyList(), ROOT_SCHEMA_NAME);
+      this.storages = storages;
+    }
+
+    @Override
+    public Set<String> getSubSchemaNames() {
+      return storages.availablePlugins();
+    }
+
+    @Override
+    public Schema getSubSchema(String name) {
```

Comment:
@vvysotskyi I aimed here to work according to the existing design as I saw it.  DynamicRootSchema is a subtype of CalciteSchema which is a wrapper around a Schema stored in a member variable.  In the case of CalciteSchema it is the wrapped Schema that implements getSubSchema and getSubSchemaNames, not the CalciteSchema wrapper.  So I reasoned that for the case of DynamicRootSchema, the wrapped RootSchema should implement, indeed override, the mentioned methods.

Reply:
I think the case of `DynamicRootSchema` is the special one, since it overrides the parent method `getImplicitSubSchema()` that uses the `Schema.getSubSchema()` method in the parent class. I think it was done intentionally, in the opposite case all the logic related to loading schemas simple would be placed in the `RootSchema`.

Reply:
@vvysotskyi good catch, thanks!